<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Introduction | Data science for biology</title>
  <meta name="description" content="Chapter 12 Introduction | Data science for biology" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Introduction | Data science for biology" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Introduction | Data science for biology" />
  
  
  

<meta name="author" content="alena savachenko" />


<meta name="date" content="2024-11-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction-1.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/d3-4.13.0/./d3/d3.v4.min.js"></script>
<link href="libs/g3viz-0.4.0/./g3viz/g3-lollipop.min.css" rel="stylesheet" />
<script src="libs/g3viz-0.4.0/./g3viz/g3-lollipop.min.js"></script>
<script src="libs/g3Lollipop-binding-1.2.0/g3Lollipop.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/css/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="assets/img/logo_toc.png"></a></li>
<li><a href="./">Data science for biology </a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="the-guerrilla-data_management-principles.html"><a href="the-guerrilla-data_management-principles.html"><i class="fa fa-check"></i><b>2</b> The Guerrilla Data_management Principles</a>
<ul>
<li class="chapter" data-level="2.1" data-path="the-guerrilla-data_management-principles.html"><a href="the-guerrilla-data_management-principles.html#portfolio-assignment-1.2-part-1"><i class="fa fa-check"></i><b>2.1</b> Portfolio assignment 1.2 Part 1</a></li>
<li class="chapter" data-level="2.2" data-path="the-guerrilla-data_management-principles.html"><a href="the-guerrilla-data_management-principles.html#portfolio-assignment-1.2-looking-ahead.-package-genvisr"><i class="fa fa-check"></i><b>2.2</b> Portfolio assignment 1.2 Looking ahead. Package GenVisR</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="agile-collaboration.html"><a href="agile-collaboration.html"><i class="fa fa-check"></i><b>3</b> Agile collaboration</a></li>
<li class="chapter" data-level="4" data-path="data-managment-principles.html"><a href="data-managment-principles.html"><i class="fa fa-check"></i><b>4</b> Data managment principles</a></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a></li>
<li class="chapter" data-level="6" data-path="reproducibility-issues-in-a-scientific-publication..html"><a href="reproducibility-issues-in-a-scientific-publication..html"><i class="fa fa-check"></i><b>6</b> Reproducibility issues in a scientific publication.</a></li>
<li class="chapter" data-level="7" data-path="reproducble-research.-part-2.html"><a href="reproducble-research.-part-2.html"><i class="fa fa-check"></i><b>7</b> Reproducble research. Part 2</a></li>
<li class="chapter" data-level="8" data-path="working-with-relational-databases.html"><a href="working-with-relational-databases.html"><i class="fa fa-check"></i><b>8</b> Working with relational databases</a></li>
<li class="chapter" data-level="9" data-path="creating-r-packages.html"><a href="creating-r-packages.html"><i class="fa fa-check"></i><b>9</b> Creating R packages</a></li>
<li class="chapter" data-level="10" data-path="parameterized-rmarkdown.html"><a href="parameterized-rmarkdown.html"><i class="fa fa-check"></i><b>10</b> parameterized RMarkdown</a></li>
<li class="chapter" data-level="11" data-path="introduction-1.html"><a href="introduction-1.html"><i class="fa fa-check"></i><b>11</b> Introduction</a></li>
<li class="chapter" data-level="12" data-path="introduction-2.html"><a href="introduction-2.html"><i class="fa fa-check"></i><b>12</b> Introduction</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/alenagrrr3/portfolio_alena_savachenko" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data science for biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction-2" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Introduction<a href="introduction-2.html#introduction-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<h3>
Publishing
</h3>
<p>Introduction</p>
<p>The present work explores data on the expressed genes in cancer tissues to reflect information about changes in the transcriptome at different stages of cancer development.MSF-7 tumor cell line were sequenced by MinION sequencer at the ILC Research Center <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<p>The aim of the study is to annotate, characterize and structure transcriptome analysis of genes expressed during the development of using RNA/cDNA nanopore sequencing technology (Oxford Nanopore Technologies).Sequencing results were compared with previously obtained results of the Ying Chen from 2021 in the cancer study.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
<p>Five cancer cell types were sequenced and the comparative organization of gene clusters was studied. Transcriptome data from long read and short read sequencers were obtained. Bioinformatic and statistical analysis of the experimental data was performed. The work demonstrated the reproducibility of the results under different experimental conditions.
<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
The purpose of this study was to obtain data on the expressed genes of tumor cells, dynamics of their transcription and polymorphism. The aim of the study included the involvement of different proteins in the regulation of proliferation canerogene cells of the MSF-7, Hct116, K562, HepG2 cell lines.<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a></p>
<p>Studies on cancer biology have been conducted throughout the 21st centuries, including genomic variation, repeat regions, phasing, and transcript isoforms <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<p>Despite the available research on tumor cell function remains open the question of the mechanisms controlling cell proliferation and “escaping” from apoptotic death. Therefore, the study of molecular-genetic and transcriptional features of development of tumor cell lines is an urgent and highly demanded direction of modern research.</p>
<p>At the same time, new technologies long-read sequencing technologies are improving the accuracy of genome annotation and detection of new transcripts,
including long non-coding RNAs (dnRNAs). Therefore, the use of third-generation sequencers to characterization of the transcriptome and the study of structural features of oncogenes is an important task, the solution of which will provide new information on the genetic basis of cancer development.<a href="#fn6" class="footnote-ref" id="fnref6"><sup>6</sup></a></p>
<p>Third-generation sequencing provided information on genes involved in the process of tumor cell development. In addition to the complexity and
polymorphism of the genes themselves involved in the tumor cell development process, their study is complicated by the genomic
organization of hybrid genes ( suppelmentory files, media-5.csv ), combines complex genomes, which makes annotation of its genome difficult.</p>
<p>A key tool for identifying genes, structural variants and characteristics is Oxford .
high-throughput Nanopore technology <a href="#fn7" class="footnote-ref" id="fnref7"><sup>7</sup></a></p>
<p>Nanopore sequencing is a technology that allows you to
direct analysis of long DNA or RNA fragments in real time.
in real time, based on the change in electrical current as nucleic acids as nucleic acids pass through protein nanopores. (application). Nanopore sequencing is used to assemble genomes, detect full-length transcripts, searching for nucleotide modifications. Applications are gaining momentum in more specialized applications, such as rapid clinical diagnostics and epidemiologic surveillance. However, in order to obtain reads of significant length, it is necessary to
optimal sample preparation <a href="#fn8" class="footnote-ref" id="fnref8"><sup>8</sup></a>.</p>
<p>A prerequisite for obtaining long
reads is the availability of high-quality high-molecular-weight DNA. If the start material is highly fragmented and degraded, it is not possible to obtain long reads. The methodology of isolation DNA should be carefully selected for each type of sample. The MinION is the most popular and most compact sequencer.<a href="#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a></p>
<p>The software decodes the sequences in real time so they can be correlated with different markers. For example, it is possible to identify a gene associated with cancer.</p>
<p><span class="citation">(<strong>GoekeLabSgnexdata2024?</strong>)</span></p>

</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p><span class="citation">(<strong>HttpsCanvashunlCourses?</strong>)</span><a href="introduction-2.html#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><span class="citation">(<strong>chenSystematicBenchmarkNanopore2021b?</strong>)</span><a href="introduction-2.html#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><span class="citation">(<strong>chenSystematicBenchmarkNanopore2021b?</strong>)</span><a href="introduction-2.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p><span class="citation">(<strong>chenSystematicBenchmarkNanopore2021b?</strong>)</span><a href="introduction-2.html#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p><span class="citation">(<strong>HttpsAstoryblokcom196663a?</strong>)</span><a href="introduction-2.html#fnref5" class="footnote-back">↩︎</a></p></li>
<li id="fn6"><p><span class="citation">(<strong>chenSystematicBenchmarkNanopore2021b?</strong>)</span><a href="introduction-2.html#fnref6" class="footnote-back">↩︎</a></p></li>
<li id="fn7"><p><span class="citation">(<strong>luOxfordNanoporeMinION2016?</strong>)</span><a href="introduction-2.html#fnref7" class="footnote-back">↩︎</a></p></li>
<li id="fn8"><p><span class="citation">(<strong>shanmuganandamLibraryPreparationProtocol2019?</strong>)</span><a href="introduction-2.html#fnref8" class="footnote-back">↩︎</a></p></li>
<li id="fn9"><p><span class="citation">(<strong>GoekeLabSgnexdata2024?</strong>)</span><a href="introduction-2.html#fnref9" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
<div id="footer">Copyright &copy;  Hogeschool Utrecht   DSFB2  <i class="fa-solid fa-webhook"></i> 2024  		</footer>




            </section>

          </div>
        </div>
      </div>
<a href="introduction-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
